### PR TITLE
Add release automation logic

### PR DESCRIPTION
**Pull request by sydney-runkle**
Closes https://github.com/pydantic/logfire/issues/470

See readme for new pattern:

```
1. Run `uv run release/prepare.py {VERSION}` from the root of the repository. This will:
    * Update the version number in the `pyproject.toml` files in the root and in `logfire-api`.
    * Add a new section to CHANGELOG.md with a title containing the version number tag and current date.
    * Add a line at the end of this section that looks something like [v1.0.1]: https://github.com/pydantic/logfire/compare/v{PREV_VERSION}...v1.0.1 but with the correct version number tags.
2. Curate the changes in CHANGELOG.md:
    * Make sure the markdown is valid; in particular, check text that should be in code-blocks is.
    * Mark any breaking changes with **Breaking Change:**.
    * Deduplicate the packaging entries to include only the most recent version bumps for each package.
3. Run `uv run release/push.py` from the root of the repository. This will:
    * Create a PR with the changes you made in the previous steps.
    * Add a label to the PR to indicate that it's a release PR.
    * Open a draft release on GitHub with the changes you made in the previous steps.
4. Review the PR and merge it.
5. Publish the release and wait for the CI to finish building and publishing the new version.
```

Sample draft release created: https://github.com/pydantic/logfire/releases/tag/untagged-f0b744b87f685d76bbd2
Sample PR (note, I've changed the base for clarity, but normally it would be against `main`): https://github.com/pydantic/logfire/pull/511

Looks like I updated stubs as well 🤷‍♀️

### CODE REVIEW

**Comment by sydney-runkle** [comment]
Sample diff from `uv run release/prepare.py 1.1.1`:

```
diff --git a/CHANGELOG.md b/CHANGELOG.md
index 2b6693c..49a8e07 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,14 @@
 # Release Notes
 
+## [v1.1.1] (2024-10-17)
+
+## What's Changed
+* Update duration percentile example with approx_percentile_cont by @alexmojaki in [#500](https://github.com/pydantic/logfire/pull/500)
+* PYD-1326: basic live view sql explanation in docs by @ChristopherGS in [#501](https://github.com/pydantic/logfire/pull/501)
+* Add `excluded_urls` example for `/` by @sydney-runkle in [#487](https://github.com/pydantic/logfire/pull/487)
+* Bump dependencies with `uv sync --upgrade` by @sydney-runkle in [#506](https://github.com/pydantic/logfire/pull/506)
+* Revert "Bump dependencies with `uv sync --upgrade`" by @sydney-runkle in [#510](https://github.com/pydantic/logfire/pull/510)
+
 ## [v1.1.0] (2024-10-14)
 
 * Fix error in checking for generators in auto-tracing by @alexmojaki in https://github.com/pydantic/logfire/pull/498
@@ -353,3 +362,4 @@ First release from new repo!
 [v1.0.0]: https://github.com/pydantic/logfire/compare/v0.55.0...v1.0.0
 [v1.0.1]: https://github.com/pydantic/logfire/compare/v1.0.0...v1.0.1
 [v1.1.0]: https://github.com/pydantic/logfire/compare/v1.0.1...v1.1.0
+[v1.1.1]: https://github.com/pydantic/logfire/compare/v1.1.0...v1.1.1
diff --git a/logfire-api/pyproject.toml b/logfire-api/pyproject.toml
index b23ccbc..f215c06 100644
--- a/logfire-api/pyproject.toml
+++ b/logfire-api/pyproject.toml
@@ -4,7 +4,7 @@ build-backend = "hatchling.build"
 
 [project]
 name = "logfire-api"
-version = "1.1.0"
+version = "1.1.1"
 description = "Shim for the Logfire SDK which does nothing unless Logfire is installed"
 authors = [
     { name = "Pydantic Team", email = "engineering@pydantic.dev" },
diff --git a/pyproject.toml b/pyproject.toml
index c948d9f..0d9c231 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,7 +4,7 @@ build-backend = "hatchling.build"
 
 [project]
 name = "logfire"
-version = "1.1.0"
+version = "1.1.1"
 description = "The best Python observability tool! 🪵🔥"
 requires-python = ">=3.8"
 authors = [
diff --git a/uv.lock b/uv.lock
index 9a0e0ec..b215e0c 100644
--- a/uv.lock
+++ b/uv.lock
@@ -1365,7 +1365,7 @@ wheels = [
 
 [[package]]
 name = "logfire"
-version = "1.1.0"
+version = "1.1.1"
 source = { editable = "." }
 dependencies = [
     { name = "executing" },
@@ -1612,7 +1612,7 @@ dev = [
 
 [[package]]
 name = "logfire-api"
-version = "1.1.0"
+version = "1.1.1"
 source = { editable = "logfire-api" }
 
 [[package]]
```

**Comment by sydney-runkle** [comment]
Sample draft release created: https://github.com/pydantic/logfire/releases/tag/untagged-f0b744b87f685d76bbd2
Sample PR (note, I've changed the base for clarity, but normally it would be against `main`): https://github.com/pydantic/logfire/pull/511

File: CHANGELOG.md [hunk 1]
```diff
@@ -311,45 +311,45 @@ First release from new repo!
 * Ensure `logfire.testing` doesn't depend on pydantic and eval_type_backport by @alexmojaki in https://github.com/pydantic/logfire/pull/40
 * Allow using pydantic plugin with models defined before calling logfire.configure by @alexmojaki in https://github.com/pydantic/logfire/pull/36
 
-[v1.1.0]: https://github.com/pydantic/logfire/compare/v1.0.1...v1.1.0
```
Comment (sydney-runkle) [inline CHANGELOG.md:314]: Reversed the order here so that we can simply append in the future

**Comment by alexmojaki** [comment]
Thank you! I haven't reviewed the code because I don't think there's really a need, if there's issues we'll discover them.

**Pull request merged**